web/ui/src/Services/AppsController.js

```diff
@@ -292,34 +267,108 @@
                 $scope.templates.data = utils.mapToArr(templates);
             });
         }
-       
-        function fetchInternalServiceInstances(app) {
-            return resourcesFactory.v2.getInternalServiceInstances(app.id).then( data => {
-                app.instances = data;
+
+        function refreshApps() {
+            return fetchApps()
+                .then(fillOutApps)
+                .then(fetchServicesHealth)
+                .then(fetchInternalServicesHealth)
+                .then(() => touch());
+        }
+
+        function fetchApps() {
+            return $q.all([fetchInternalService(), fetchServices()])
+                .then(results => {
+                    $scope.apps = [];
+
+                    var isvc = results[0];
+                    $scope.apps.push(isvc);
+
+                    var services = results[1];
+                    services.forEach(s => {
+                        $scope.apps.push(s);
+                    });
+                });
+        }
+
+        function fetchChildEndpoints(app) {
+            return resourcesFactory.v2.getServiceChildPublicEndpoints(app.id).then(data => {
+                app.publicEndpoints = data;
             });
         }
 
-        function getInternalServicesHealth() {
+        function fillOutApps() {
+            let promises = [];
+            $scope.apps.forEach(a => {
+                if (a.isIsvc()) {
+                    promises.push(() => a.fetchInstances());
+                } else {
+                    promises.push(fetchChildEndpoints(a));
+                }
+            });
+
+            return $q.all(promises);
+        }
+
+        function fetchServices() {
+            let deferred = $q.defer();
+            resourcesFactory.v2.getTenants().then( data => {
+                var services = data.map(s => new Service(s));
+                deferred.resolve(services);
```

Comment:
above 2 lines need to be indented one more or something... its not quite right around here.